protocol/messenger_builder_test.go

```diff
@@ -90,10 +94,37 @@ func newTestMessenger(waku types.Waku, config testMessengerConfig, extraOptions
 		return nil, err
 	}
 
+	if config.unhandledMessagesTracker != nil {
+		m.unhandledMessagesTracker = config.unhandledMessagesTracker.addMessage
+	}
+
 	err = m.Init()
 	if err != nil {
 		return nil, err
 	}
 
 	return m, nil
 }
+
+type unhandedMessage struct {
+	*v1protocol.StatusMessage
+	err error
+}
+
+type unhandledMessagesTracker struct {
+	messages map[protobuf.ApplicationMetadataMessage_Type][]*unhandedMessage
+}
+
+func (u *unhandledMessagesTracker) addMessage(msg *v1protocol.StatusMessage, err error) {
+	msgType := msg.ApplicationLayer.Type
+
+	if _, exists := u.messages[msgType]; !exists {
+		u.messages[msgType] = []*unhandedMessage{}
+	}
+
+	newMessage := &unhandedMessage{
+		StatusMessage: msg,
+		err:           err,
+	}
+	u.messages[msgType] = append(u.messages[msgType], newMessage)
+}
```

Comment:
Not a lot of code, but maybe move this to a separate file?
It would also make sense to use `unhandedMessage` as an argument in `addMessage`

Reply:
> Not a lot of code, but maybe move this to a separate file?

Was thinking about it but wasn't able to come up with proper name :laughing: This `_test.go` suffix makes test utilities very hard to name. I will try to come up with something in follow-up maybe?

> It would also make sense to use `unhandedMessage` as an argument in `addMessage`

I can't, `addMessage` must fulfill the callback interface defined in Messenger::unhandledMessagesTracker.